kernel/aster-nix/Cargo.toml

```diff
@@ -70,6 +69,13 @@ bytemuck_derive = "1.5.0"
 takeable = "0.2.2"
 cfg-if = "1.0"
 
+[target.x86_64-unknown-none.dependencies]
+aster-time = { path = "../comps/time" }
+
+[target.riscv64gc-unknown-none-elf.dependencies]
+aster-goldfish = { path = "../comps/goldfish" }
```

Comment:
I'm not sure if there are other reasons for introducing the `goldfish` crate. However, from my perspective, the main reason is that the RTC part requires different implementations based on the architecture. But why can't this differentiation be handled in the original component using `cfg`? Since the public interfaces have not been changed.

Reply:
I'm not sure if adding all implementations of RTC to `aster-time` would make `aster-time` inflate. Because there are different implementations on different platform on RISC-V, and `goldfish` is only one of them that is mostly used in VM.

In other words, I'm not sure if it's better to have a mono `aster-time` crate or to have differents crates for differents drivers (and extract the common code to a `rtc` prerequisite library), like `aster-cmos`, `aster-goldfish`, and etc.

Reply:
I think that implementing RTC support for different platforms in the same location is better, since it can achieve more code reusing. After all, the logic for things like `tsc` and `clocksource` is consistent. We can implement the differentiated parts in different files and provide a unified interface.

Reply:
I think both approaches are viable. The question is if the RTC drivers on x86 and on RISC-V can share a lot of code. If so, they should be merged into one crate. If not, they can be two parallel crates. Either way, you need to make the crate name more self-describing.

For the first approach, there will be only one crate, which is `time`. Plain and simple.

For the second approach, there should be two crates, whose names indicate their purposes. Two names that come to my mind are: `time-x86` and `time-goldfish`.

Since the data structures in `aster-goldfish` resemble those in `time`, I guess that it is more preferable to merge `aster-goldfish` into the `time` crate.